hook!-full-game/src/Player/Player.gd

```diff
@@ -8,17 +8,23 @@ onready var floor_detector: RayCast2D = $FloorDetector
 onready var skin: Position2D = $Skin
 onready var skills: Node = $Skills
 onready var stats: Stats = $Stats
+onready var animation_player : AnimationPlayer = $AnimationPlayer
 
 var _info_dict: = {} setget _set_info_dict
 
 var _gravity: = 0.0
 var _velocity: = Vector2.ZERO
 
 var _state: = "idle"
+var checkpoints := []
+var dead := false
```

Comment:
There's already an `active` property on the player, and you can check which state the `StateMachine` is in instead of adding an extra boolean